modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -378,20 +378,20 @@ private void onEvent(CoreEvent event, ExecutorCallback executorCallback) {
     try {
       SdkInternalContext sdkInternalContext = from(event);
 
-      final Optional<ConfigurationInstance> configuration = sdkInternalContext.getConfiguration();
-      final Map<String, Object> resolutionResult = sdkInternalContext.getResolutionResult();
+      final Optional<ConfigurationInstance> configuration = sdkInternalContext.getConfiguration(getLocation());
+      final Map<String, Object> resolutionResult = sdkInternalContext.getResolutionResult(getLocation());
 
       OperationExecutionFunction operationExecutionFunction = (parameters, operationEvent, callback) -> {
-        sdkInternalContext.setOperationExecutionParams(configuration, parameters, operationEvent, callback);
+        sdkInternalContext.setOperationExecutionParams(getLocation(), configuration, parameters, operationEvent, callback);
 
         fluxSupplier.get().next(operationEvent);
       };
 
       if (getLocation() != null) {
         ((DefaultFlowCallStack) event.getFlowCallStack())
             .setCurrentProcessorPath(resolvedProcessorRepresentation);
-        sdkInternalContext.getPolicyToApply().process(event, operationExecutionFunction, () -> resolutionResult, getLocation(),
-                                                      executorCallback);
+        sdkInternalContext.getPolicyToApply(getLocation())
```

Comment:
IIRC, getLocation() was kind of expensive. Since already invoked in line 390, I would extract to a variable

Reply:
was, it os now caches from some weeks ago